### PR TITLE
[Hanging Garden] - feat: Virtualization of garden

### PR DESCRIPTION
**Pull request by jaysencpp**
# Description

Still WIP

Completes: AB#FILL_IN_YOUR_ISSUE_ID

## Checklist:

-   [ ] I have performed a self-review of my own code.
-   [ ] I have linked my DevOps task using the AB# tag.
-   [ ] My code is easy to read, and comments are added where needed.
-   [ ] My code is covered by tests.

### CODE REVIEW

File: src/Core/WorkSpace/src/Tabs/GardenTab.tsx [hunk 1]
```diff
@@ -3,8 +3,8 @@ import { Garden } from '../../../../components/ParkView/Components/Garden';
 import { useDataContext } from '../Context/DataProvider';
 
 export const GardenTab = (): JSX.Element => {
-    const { data } = useFilteredData();
+    const { data, isLoading } = useFilteredData();
     const { gardenOptions } = useDataContext();
-
+    if (isLoading) return <h1>Loading</h1>;
```
Comment (jaysencpp) [inline src/Core/WorkSpace/src/Tabs/GardenTab.tsx:8]: Handle this better.

File: src/apps/WorkOrder/Garden/components/WorkOrderItem/styles.ts [hunk 2]
```diff
@@ -2,17 +2,21 @@ import styled from 'styled-components';
 import { Item } from '../../../../../components/ParkView/Styles/item';
 export type WorkOrderItemProps = { backgroundColor: string; textColor: string; background: string };
 //@ts-ignore
+/* position: relative; */
 export const WorkOrderWrapper = styled(Item)<WorkOrderItemProps>`
-    position: relative;
-    display: flex;
+    position: absolute;
```
Comment (jaysencpp) [inline src/apps/WorkOrder/Garden/components/WorkOrderItem/styles.ts:7]: pos relative remove z index

File: src/apps/WorkOrder/Garden/components/WorkOrderItem/test.css [hunk 3]
```diff
@@ -0,0 +1,13 @@
+.circles {
```
Comment (jaysencpp) [inline src/apps/WorkOrder/Garden/components/WorkOrderItem/test.css:1]: remove

File: src/components/ParkView/Components/Garden.tsx [hunk 4]
```diff
@@ -2,17 +2,19 @@ import { ParkViewProvider } from '../Context/ParkViewProvider';
...
 
 interface GardenProps<T> {
     data: T[];
     gardenOptions: GardenOptions<T> | undefined;
 }
 export function Garden<T>({ gardenOptions, data }: GardenProps<T>): JSX.Element {
     if (!gardenOptions) return <NoGardenOptions />;
-
+    //TODO:Handle no data better in garden
+    if (!data) return <h1>No data</h1>;
```
Comment (jaysencpp) [inline src/components/ParkView/Components/Garden.tsx:14]: return null

File: src/components/ParkView/Components/VirtualGarden/Container.tsx [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+            if (amountOfColumns > 0) {
+                const width = (itemWidth && itemWidth(garden, gardenKey.toString())) || 300;
+                setWidths(new Array(amountOfColumns).fill(width));
+            }
+        }
+    }, [amountOfColumns, itemWidth]);
+
+    //TODO: Handle widths = 0 better
+    if (widths.length === 0) {
+        return <h1>Length 0</h1>;
```
Comment (jaysencpp) [inline src/components/ParkView/Components/VirtualGarden/Container.tsx:69]: returnnull

File: src/components/ParkView/Components/VirtualGarden/GardenItemContainer.tsx [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+type VirtualHookReturn = Pick<ReturnType<typeof useVirtual>, 'virtualItems' | 'scrollToIndex'>;
+type PackageContainerProps<T> = {
+    virtualColumn: VirtualItem;
+    rowVirtualizer: VirtualHookReturn;
+    garden: GardenGroups<T>;
+    sortData?: (data: T[], ...groupByKeys: (keyof T)[]) => T[];
+    itemKey: keyof T;
+    fieldSettings?: FieldSettings<T>;
+    packageChild: React.MemoExoticComponent<(args: CustomItemView<T>) => JSX.Element>;
+    handleExpand: any;
```
Comment (jaysencpp) [inline src/components/ParkView/Components/VirtualGarden/GardenItemContainer.tsx:28]: type handleexpand

File: src/components/ParkView/Components/VirtualGarden/VirtualGarden.tsx [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+        <Layout
+            rowTotalSize={rowVirtualizer.totalSize}
+            columnTotalSize={columnVirtualizer.totalSize}
+            parentRef={parentRef}
+            isScrolling={isScrolling}
+        >
+            <HeaderContainer<T>
+                columnVirtualizer={columnVirtualizer}
+                garden={sortedColumns}
+                headerChild={headerChild!}
```
Comment (jaysencpp) [inline src/components/ParkView/Components/VirtualGarden/VirtualGarden.tsx:101]: headerchild can be undefined

File: src/components/ParkView/Components/VirtualGarden/VirtualGarden.tsx [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+
+                return (
+                    <Fragment key={virtualColumn.index}>
+                        <GardenItemContainer
+                            rowVirtualizer={rowVirtualizer}
+                            garden={sortedColumns}
+                            virtualColumn={virtualColumn}
+                            sortData={sortData}
+                            itemKey={itemKey}
+                            packageChild={packageChild!}
```
Comment (jaysencpp) [inline src/components/ParkView/Components/VirtualGarden/VirtualGarden.tsx:116]: packagechild can be undefined

File: src/components/ParkView/Styles/group.ts [hunk 9]
```diff
@@ -9,7 +9,7 @@ export const SubGroup = styled.div`
 `;
 
 export const Pack = styled.div`
-    padding: 0.5rem 1rem;
+    /* padding: 0.5rem 1rem; */
```
Comment (jaysencpp) [inline src/components/ParkView/Styles/group.ts:12]: remove comment and overwrite padding where its inherited

**Review by Gustav-Eikaas — APPROVED** [review]

**Pull request merged**